File: planner/app/src/main/java/com/sunilson/firenote/adapters/ColorVisibilityAdapter.java
package com.sunilson.firenote.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;

import com.sunilson.firenote.Interfaces.MainActivityInterface;
import com.sunilson.firenote.R;
import com.sunilson.firenote.presentation.views.ColorElementView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by linus_000 on 17.11.2016.
 */

public class ColorVisibilityAdapter extends ArrayAdapter<NoteColor> {

    private List<NoteColor> list = new ArrayList<>();

    public ColorVisibilityAdapter(Context context, int resource) {
        super(context, resource);
    }

    @Override
    public void add(NoteColor object) {
        list.add(object);
        notifyDataSetChanged();
    }

    @Nullable
    @Override
    public NoteColor getItem(int position) {
        return list.get(position);
    }

    @Override
    public int getCount() {
        return list.size();
    }

    /**
     * Uncheck all elements of the Adapter
     */
    public void uncheckAll() {
        MainActivityInterface mainActivityInterface = (MainActivityInterface) ((BaseApplication) getContext().getApplicationContext()).mainContext;

        //Iterate over list
        for (int i = 0; i < getCount(); i++) {
            LocalSettingsManager.getInstance().setColorVisibility((getItem(i)).getColor(), -1);
            mainActivityInterface.getElementAdapter().hideElements();
        }

        notifyDataSetChanged();
    }

    /**
     * Check all elements of the Adapter
     */
    public void checkAll() {
        MainActivityInterface mainActivityInterface = (MainActivityInterface) ((BaseApplication) getContext().getApplicationContext()).mainContext;

        //Iterate over list
        for (int i = 0; i < getCount(); i++) {
            LocalSettingsManager.getInstance().setColorVisibility((getItem(i)).getColor(), 1);
            mainActivityInterface.getElementAdapter().hideElements();
        }

        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        ColorElementView row = (ColorElementView) convertView;

        ColorVisibilityAdapter.ElementHolder elementHolder;

        if(row == null) {
            row = new ColorElementView(getContext());
            elementHolder = new ColorVisibilityAdapter.ElementHolder();
            elementHolder.icon = (ImageView) row.findViewById(R.id.color_list_icon);
            row.setTag(elementHolder);
        } else {
            elementHolder = (ElementHolder) row.getTag();
        }

        NoteColor noteColor = getItem(position);
        row.setBackgroundColor(noteColor.getColor());

        elementHolder.icon.setImageResource(R.drawable.ic_cancel_white_24dp);

        if (LocalSettingsManager.getInstance().getColorVisibility(noteColor.getColor()) == -1) {
            row.setChecked(true);
        } else {
            row.setChecked(false);
        }

        return row;
    }

    static class ElementHolder {
        public ImageView icon;
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/adapters/ColorVisibilityAdapter.kt
package com.sunilson.firenote.presentation.adapters

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.sunilson.firenote.Interfaces.MainActivityInterface
import com.sunilson.firenote.R
import com.sunilson.firenote.data.models.NoteColor
import com.sunilson.firenote.presentation.shared.base.adapters.BaseArrayAdapter
import com.sunilson.firenote.presentation.shared.interfaces.HasElementList
import com.sunilson.firenote.presentation.shared.singletons.LocalSettingsManager
import com.sunilson.firenote.presentation.shared.views.ColorElementView
import javax.inject.Inject

class ColorVisibilityAdapter(context: Context, val localSettingsManager: LocalSettingsManager) : BaseArrayAdapter<NoteColor>(context, R.layout.color_list_layout), CheckableArrayAdapter {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var row = convertView as ColorElementView?

        val elementHolder =  if(row == null) {
            row = ColorElementView(context)
            val tempHolder = ColorVisibilityAdapter.ElementHolder(row.findViewById(R.id.color_list_icon))
            row.tag = tempHolder
            tempHolder
        } else {
            row.tag as ColorVisibilityAdapter.ElementHolder
        }

        row.tag = elementHolder

        return row
    }

    override fun toggleAll(checked: Boolean) {
        data.forEach {
            localSettingsManager.setColorVisibility(it.color, if (checked) -1 else 1)
            (context as HasElementList).adapter.hideElements()
        }
        notifyDataSetChanged()
    }

    private class ElementHolder (val icon : ImageView)
}